client/dev-tools/components/dev-tools.tsx

```diff
@@ -170,7 +172,14 @@ const DevTools = () => {
 					</>
 				) : (
 					<>
-						<Button variant="primary" className="dev-tools__button" href={ upgradeLink }>
+						<Button
+							variant="primary"
+							className="dev-tools__button"
+							href={ upgradeLink }
+							onClick={ () =>
+								dispatch( recordTracksEvent( 'calypso_hosting_settings_upgrade_plan_click' ) )
```

Comment:
Let's wait for a bit until we settle on the tab name here: https://github.com/Automattic/dotcom-forge/issues/7590

Reply:
Let's use `hosting_features` instead of `hosting_settings` (https://github.com/Automattic/dotcom-forge/issues/7641)